randovania/data/json_data/prime3.txt

```diff
@@ -1575,25 +1843,42 @@ Asset id: 12448241480574438186
 > Dock to Hangar Bay Hall; Heals? False
   * Normal Door to Hangar Bay Hall/Dock to Hangar Bay
   > Dock to Ruined Shrine
-      Trivial
+      After Hangar Bay Power
 
 > Dock to Ruined Shrine; Heals? False
   * Normal Door to Ruined Shrine/Dock to Hangar Bay
   > Dock to Hangar Bay Hall
-      Trivial
+      After Hangar Bay Power
   > Pickup (Energy Cell)
-      Grapple Lasso or Use Screw Attack (Space Jump)
+      Trivial
+  > Pickup (Main Ship Missiles)
+      After Hangar Bay Ceiling
+  > Event - Hangar Ceiling
+      Grapple Swing or Use Screw Attack (Space Jump)
 
 > Pickup (Energy Cell); Heals? False
   * Pickup 42; Major Location? False
-  > Dock to Ruined Shrine
-      Trivial
-  > Pickup (Main Ship Missiles)
+  > Event - Power
       Trivial
 
 > Pickup (Main Ship Missiles); Heals? False
   * Pickup 43; Major Location? True
-  > Pickup (Energy Cell)
+  > Event - Ship Upgrade
+      Trivial
+
+> Event - Power; Heals? False
+  * Event Hangar Bay Power
+  > Dock to Ruined Shrine
+      Trivial
+
+> Event - Hangar Ceiling; Heals? False
+  * Event Hangar Bay Ceiling
+  > Dock to Ruined Shrine
+      Trivial
+
+> Event - Ship Upgrade; Heals? False
+  * Event Hangar Bay Ship Upgrade
+  > Dock to Ruined Shrine
       Trivial
 
```

Comment:
Add another node to this room called "Samus Ship," with "Heals?" checked. Have it connect to Ruined Shrine.